x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/FieldNameTranslators.java

```diff
@@ -68,6 +70,19 @@ public final class FieldNameTranslators {
         )
     );
 
+    public static final FieldNameTranslators ROLE_FIELD_NAME_TRANSLATORS = new FieldNameTranslators(
+        List.of(
+            idemFieldNameTranslator("name"),
+            idemFieldNameTranslator("description"),
+            idemFieldNameTranslator("applications.application"),
+            idemFieldNameTranslator("applications.resources"),
+            idemFieldNameTranslator("applications.privileges"),
```

Comment:
It irks me a little that `application` is singular here but `resources` and `privileges` are plurals.
However, I think this is preferable because these are the names of these fields in the response.
WDYT @jfreden , should we make these all singulars?

Reply:
Yes, I agree, that's annoying. 

We can't change the current json serialization of a role response for backwards compatibility reasons and having it different for query roles doesn't make sense. 

I think it's more important that it's consistent with the response, so keeping `application` singular.